control-plane-operator/hostedclusterconfigoperator/controllers/drainer/setup.go

```diff
@@ -0,0 +1,35 @@
+package drainer
+
+import (
+	"fmt"
+
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/operator"
+	corev1 "k8s.io/api/core/v1"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+func Setup(opts *operator.HostedClusterConfigOperatorConfig) error {
+	targetKubeClient, err := kubeclient.NewForConfig(opts.TargetConfig)
+	if err != nil {
+		return err
+	}
+	r := &Reconciler{
+		client:                 opts.CPCluster.GetClient(),
+		guestClusterClient:     opts.Manager.GetClient(),
+		guestClusterKubeClient: targetKubeClient,
```

Comment:
nit: worth a comment why we need both.

Reply:
added to the struct definition (and squashed to the previous commit)